neo/Network/P2P/TaskManager.cs

```diff
@@ -54,16 +56,21 @@ private void OnNewTasks(InvPayload payload)
             HashSet<UInt256> hashes = new HashSet<UInt256>(payload.Hashes);
             hashes.ExceptWith(knownHashes);
             if (payload.Type == InventoryType.Block)
-                session.AvailableTasks.UnionWith(hashes.Where(p => globalTasks.Contains(p)));
-            hashes.ExceptWith(globalTasks);
+                session.AvailableTasks.UnionWith(hashes.Where(p => globalTasks.ContainsKey(p)));
+
+            hashes.ExceptWith(globalTasks.Keys);
             if (hashes.Count == 0)
             {
                 RequestTasks(session);
                 return;
             }
-            globalTasks.UnionWith(hashes);
+            
             foreach (UInt256 hash in hashes)
+            {
+                globalTasks[hash] = 1;
```

Comment:
`globalTasks[hash]++`?

Reply:
Men, eagle eyes, @erikzhang.
It looks like it should be ++.

Reply:
= 1 means start at 1, it will still allow 3 block tasks per hash.